packages/drivers/local-driver/src/localDocumentStorageService.ts

```diff
@@ -99,24 +99,29 @@ export class LocalDocumentStorageService implements IDocumentStorageService {
 		const rawTree = await this.manager.getTree(versionId);
 		const snapshotTree = buildGitTreeHierarchy(rawTree, this.blobsShaCache, true);
 		const groupIds = new Set<string>(snapshotFetchOptions?.loadingGroupIds ?? []);
+		// Always keep the .metadata blob.
```

Comment:
This is particularly interesting, so we expect to keep the metadata blob for all our nodes?

Reply:
Doesn't this blob sit at the root of the tree?

Reply:
This would just be the .metadata blob on the root which contains the attributes blob.

Reply:
I have a feeling we can really simplify this logic down. I'm of the opinion that we should just rewrite most of this logic.

Basically I wrote this logic since I actually was stripping parts of the tree away.

The algorithm now should be pretty simple. Just search through all the trees, find the blobIds to keep and then when we call populate blobIds, we just populate those blobIds instead of going through the tree.

Reply:
Simplified it as now I am populating required blob contents while we are traversing the tree. We still need to strip trees in case of group snapshot fetch.